File: Day13/day13-segregate-0s-and-1s.js
/*
Segregate 0s and 1s (Easy)
(See Java file for full problem description)
*/

class Solution {
  segregate0and1(arr) {
    let left = 0;
    for (let i = 0; i < arr.length; i++) {
      if (arr[i] === 0) {
        // Using array destructuring for a clean swap
        [arr[i], arr[left]] = [arr[left], arr[i]];
        left++;
      }
    }
  }
}

File: Day13/day13-segregate-0s-and-1s.java
/*
Segregate 0s and 1s (Easy)

Given an array arr consisting of only 0's and 1's in random order. Modify the array in-place to segregate 0s onto the left side and 1s onto the right side of the array.
*/

class Solution {
    void segregate0and1(int[] arr) {
        int left = 0;
        
        for(int i = 0; i < arr.length; i++){
            if(arr[i] == 0){
                int temp = arr[i];
                arr[i] = arr[left];
                arr[left] = temp;
                left++;
            }
        }
    }
}